pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -546,19 +561,32 @@ public SegmentMetadata getSegmentMetadata() {
     for (FieldSpec fieldSpec : _physicalFieldSpecs) {
       physicalColumnNames.add(fieldSpec.getName());
     }
-
-    return physicalColumnNames;
+    // We should include newly added columns in the physical columns
```

Comment:
What if newly introduced column is a virtual column?

Reply:
Good question. Updated the logic to handle virtual column